outputs/kubernetes.go

```diff
@@ -53,22 +60,45 @@ func (k *KubernetesClient) Init() error {
 	return nil
 }
 
+func (k KubernetesClient) prepareInputs(input map[string]string) map[string]map[string]string {
+	a := make(map[string]map[string]string)
+
+	for key, m := range k.KubeActions {
+		for id, val := range m {
+			var calcVal string
+			if strings.HasPrefix(val, regoInputPrefix) {
+				if ok := json.Valid([]byte(input["description"])); ok { // input is json
+					calcVal = gjson.Get(input["description"], strings.TrimPrefix(val, regoInputPrefix+".")).String()
+				} else {
+					calcVal = input["description"] // input is a string
+				}
+			} else {
+				calcVal = val // no rego to parse
+			}
+			a[key] = map[string]string{id: calcVal}
+		}
+	}
+
+	return a
+}
+
 func (k KubernetesClient) Send(m map[string]string) error {
 	ctx := context.Background()
+	actions := k.prepareInputs(m)
 
 	// TODO: Allow configuring of resource {pod, ds, ...}
 	pods, _ := k.clientset.CoreV1().Pods(k.KubeNamespace).List(ctx, metav1.ListOptions{
 		LabelSelector: k.KubeLabelSelector,
 	})
 	for _, pod := range pods.Items {
-		if len(k.KubeActions["labels"]) > 0 {
+		if len(actions["labels"]) > 0 {
 			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 				pod, err := k.clientset.CoreV1().Pods(pod.GetNamespace()).Get(ctx, pod.Name, metav1.GetOptions{})
 				if err != nil {
 					return fmt.Errorf("failed to get updated pod for labeling: %s, err: %w", pod.Name, err)
 				}
 
-				labels := updateMap(pod.GetLabels(), k.KubeActions["labels"])
+				labels := updateMap(pod.GetLabels(), actions["labels"])
```

Comment:
in the case of actions, we have the default keys: labels and annotations. Don't be better to use const to refer them?

Reply:
Yeah good point, thanks - I've updated that.